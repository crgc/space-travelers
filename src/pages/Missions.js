import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Table } from 'react-bootstrap';
import { getMissions, join, leave } from '../redux/missions/missionsActions';
import Mission from '../components/missions/MissionsDisplay';

const Missions = () => {
  const store = useSelector((state) => state.missionsReducer);
  const dispatch = useDispatch();
  useEffect(() => {
    if (store.length === 0) {
      dispatch(getMissions());
    }
  }, []);

  const joinMission = (id) => {
    dispatch(join(id));
  };

  const leaveMission = (id) => {
    dispatch(leave(id));
  };
  const missionsList = store;

  return (
    <Table>
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
        </tr>
      </thead>
      <tbody>
        {missionsList.map((mission) => (
          <Mission
            mission={mission}
            key={mission.mission_id}
            joinMission={joinMission}
            leaveMission={leaveMission}
          />
        ))}

      </tbody>
    </Table>
  );
};

export default Missions;
