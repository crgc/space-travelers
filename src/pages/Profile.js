import { useSelector } from 'react-redux';
import MyItems from '../components/profile/MyItems';

const Profile = () => {
  const storeMissions = useSelector((state) => state.missionsReducer);
  const rockets = useSelector((state) => state.rocketsReducer);

  const missionsReverved = storeMissions.filter(((mission) => mission.reserved))
    .map((mission) => ({
      id: mission.mission_id,
      name: mission.mission_name,
    }));

  const rocketsReserved = rockets.filter((rocket) => rocket.reserved);
  return (
    <div className="d-flex flex-row justify-content-around mt-3 profile-container">
      <MyItems title="My Missions" items={missionsReverved} />
      <MyItems title="My Rockets" items={rocketsReserved} />
    </div>
  );
};

export default Profile;
