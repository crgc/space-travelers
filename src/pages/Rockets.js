import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Rocket from '../components/rockets/Rocket';
import { getRockets } from '../redux/rockets/rockets';

const Rockets = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getRockets());
  }, []);

  const rockets = useSelector((state) => state.rocketsReducer.rockets);

  return (
    <div className="rockets-container mt-4">
      {
        rockets.map((rocket) => (
          <Rocket
            id={rocket.id}
            key={rocket.id}
            name={rocket.name}
            imageUrl={rocket.image_url}
            description={rocket.description}
          />
        ))
      }
    </div>
  );
};

export default Rockets;
