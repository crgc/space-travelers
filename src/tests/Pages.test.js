import React from 'react';
import renderer from 'react-test-renderer';
import Rockets from '../pages/Rockets'
import Missions from '../pages/Missions'
import Profile from '../pages/Profile'
import { Provider } from 'react-redux';
import store from '../redux/configureStore';

/* describe('Pages render correctly', () => {
  it('when Rockets page is rendered', () => {
    const tree = renderer
      .create(<Provider store={store}><Rockets /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('when Missions page is rendered', async () => {
    const tree = renderer
      .create(<Provider store={store}><Missions /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('when Profile page is rendered', () => {
    const tree = renderer
      .create(<Provider store={store}><Profile /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
}); */

it('when Rockets page is rendered', () => {
  const tree = renderer
    .create(<Provider store={store}><Rockets /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});