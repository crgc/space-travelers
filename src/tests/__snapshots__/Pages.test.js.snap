// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Rockets page is rendered 1`] = `
<div
  className="rockets-container mt-4"
/>
`;
